src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -638,7 +638,7 @@ public void createSnapshotWithProvidedIds(
     public void grantReadAccessToSnapshot(Snapshot snapshot, Collection<String> policies) throws InterruptedException {
         // TODO: When we support multiple datasets per snapshot, this will need to be reworked
         grantReadAccessWorker(
-            bigQueryProjectForDataset(snapshot.getFirstSnapshotSource().getDataset()),
+            bigQueryProjectForSnapshot(snapshot),
```

Comment:
Does this change fix the `TODO` too?

Reply:
This function updates ACLs on the snapshot's BigQuery dataset so it seems like the TODO isn't relevant here

Reply:
Yes, I think the original source of the comment was because we were calling `getFirstSnapshotSource()` which doesn't support multiple datasets per snapshot.